Objective-C/CBLReplicator.mm

```diff
@@ -358,6 +358,67 @@ - (void) removeChangeListenerWithToken: (id<CBLListenerToken>)token {
     }
 }
 
+- (NSSet<NSString*>*) pendingDocumentIds: (NSError**)error {
+    if (_config.replicatorType > 1) {
+        if (error)
+            *error = [NSError errorWithDomain: CBLErrorDomain
+                                         code: CBLErrorUnsupported
+                                     userInfo: @{NSLocalizedDescriptionKey: kCBLErrorMessagePullOnlyPendingDocIDs}];
+        return nil;
+    }
+    
+    C4Error c4err = {};
+    if (![self _setupC4Replicator: &c4err]) {
+        convertError(c4err, error);
+        CBLWarnError(Sync, @"%@: Replicator cannot be created: %d/%d", self, c4err.domain, c4err.code);
+        return nil;
+    }
+    
+    C4SliceResult result = c4repl_getPendingDocIDs(_repl, &c4err);
+    if (c4err.code > 0) {
+        convertError(c4err, error);
+        CBLWarnError(Sync, @"Error while fetching pending documentIds: %d/%d", c4err.domain, c4err.code);
+        return nil;
+    }
+
+    if (result.size <= 0)
+        return [NSSet set];
+
+    FLValue val = FLValue_FromData(C4Slice(result), kFLTrusted);
+    NSArray<NSString*>* list = FLValue_GetNSObject(val, nullptr);
+
+    return [NSSet setWithArray: list];
+}
+
+- (BOOL) isDocumentPending: (NSString*)documentID error: (NSError**)error {
+    CBLAssertNotNil(documentID);
+
+    if (_config.replicatorType > 1) {
+        if (error)
+            *error = [NSError errorWithDomain: CBLErrorDomain
+                code: CBLErrorUnsupported
+            userInfo: @{NSLocalizedDescriptionKey: kCBLErrorMessagePullOnlyPendingDocIDs}];
+        return NO;
+    }
+
+    C4Error c4err = {};
+    if (![self _setupC4Replicator: &c4err]) {
+        convertError(c4err, error);
+        CBLWarnError(Sync, @"%@: Replicator cannot be created: %d/%d", self, c4err.domain, c4err.code);
+        return NO;
+    }
+    
+    CBLStringBytes docID(documentID);
+    BOOL isPending = c4repl_isDocumentPending(_repl, docID, &c4err);
```

Comment:
Question: Does this method need be called under the database's lock?

Reply:
same, https://github.com/couchbase/couchbase-lite-core/blob/c6f8346ea925b7ba60200ced92e3b6770d34d54c/Replicator/c4Replicator.hh#L146